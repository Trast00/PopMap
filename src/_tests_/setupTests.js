import '@testing-library/jest-dom';
import fetchMock from 'fetch-mock';

export const testListCountry = [
  {
    name: 'test country 1',
    population: 1000,
    flags: {
      svg: 'test image url',
    },
    numericCode: 'test1',
  },
  {
    name: 'test country 2',
    population: 2000,
    flags: {
      svg: 'test image url',
    },
    numericCode: 'test2',
  },
];

export const fetchDataMock = () => {
  fetchMock.getOnce('https://restcountries.com/v2/all', {
    headers: { 'content-type': 'application/json' },
    body: testListCountry,
  });
};
