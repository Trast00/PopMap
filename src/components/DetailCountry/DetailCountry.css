.detail .country {
  padding: 16px;
  flex-direction: row;
  background-color: var(--color-primary);
}

.detail .country h4 {
  font-size: 1.4rem;
}

.detail .country .detail-wrapper {
  align-items: flex-start;
  text-align: start;
  padding-left: 24px;
}

.detail .country p {
  font-size: 1rem;
}

.sub-detail {
  text-align: start;
  padding-left: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
}

.list-details li {
  font-weight: 500;
  background-color: #db4680;
}

.list-details li:nth-child(even) {
  background-color: #cf4378;
}

.details {
  width: 100%;
  padding: 16px;
  justify-content: space-between;
}

.detail .keys {
  color: #89264c;
}
