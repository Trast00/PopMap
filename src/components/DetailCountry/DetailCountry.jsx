import React from 'react';
import './DetailCountry.css';
import PropTypes from 'prop-types';
import Country from '../ListContry/Country';

function DetailCountry(props) {
  const { country } = props;

  const listDetail = [];
  Object.keys(country).map((key) => {
    if ((typeof country[key] === 'string'
      || typeof country[key] === 'number')
      && (key !== 'alpha2Code'
        && key !== 'alpha3Code'
        && key !== 'flag'
      )) {
      return (
        listDetail.push({ key, value: country[key] })
      );
    }
    return '';
  });
  return (
    <div className="max-width column detail">
      <Country country={country} detailed />
      <div className="sub-detail">COUNTRY DETAILS: </div>
      <ul className="list-details">
        {listDetail.map((detail) => (
          <li key={detail.key} className="row details">
            <span className="values">{detail.value}</span>
            <span className="keys">{detail.key}</span>
          </li>
        ))}
      </ul>
    </div>
  );
}

DetailCountry.defaultProps = {
  country: {},
};

DetailCountry.propTypes = {
  country: PropTypes.shape({
    name: PropTypes.string,
    population: PropTypes.number,
    flags: PropTypes.shape({
      svg: PropTypes.string,
    }),
  }),
};

export default DetailCountry;
