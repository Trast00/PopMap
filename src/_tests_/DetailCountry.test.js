import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import { testListCountry } from './setupTests';
import DetailCountry from '../components/DetailCountry/DetailCountry';

describe('test render', () => {
  test('List Rocket should match snapshoot', () => {
    const tree = renderer.create(
      <React.StrictMode>
        <DetailCountry country={testListCountry[0]} />
      </React.StrictMode>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
