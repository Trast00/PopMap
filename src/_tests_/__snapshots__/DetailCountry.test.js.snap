// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render List Rocket should match snapshoot 1`] = `
<div
  className="max-width column detail"
>
  <div
    className="flex-center column country"
  >
    <img
      alt="test country 1Country Images"
      src="test image url"
    />
    <div
      className="column detail-wrapper"
    >
      <h4>
        test country 1
      </h4>
      <p>
        Citizens: 
         
        1000
      </p>
    </div>
  </div>
  <div
    className="sub-detail"
  >
    COUNTRY DETAILS: 
  </div>
  <ul
    className="list-details"
  >
    <li
      className="row details"
    >
      <span
        className="values"
      >
        test country 1
      </span>
      <span
        className="keys"
      >
        name
      </span>
    </li>
    <li
      className="row details"
    >
      <span
        className="values"
      >
        1000
      </span>
      <span
        className="keys"
      >
        population
      </span>
    </li>
    <li
      className="row details"
    >
      <span
        className="values"
      >
        test1
      </span>
      <span
        className="keys"
      >
        numericCode
      </span>
    </li>
  </ul>
</div>
`;
